aiohttp/http_writer.py

```diff
@@ -31,6 +32,8 @@ def __init__(self, protocol, transport, loop):
         self._compress = None
         self._drain_waiter = None
 
+        self.on_chunk_sent = Signal(self)
```

Comment:
Again pass traces instead of creating a new internal-only signal.

Reply:
Well, please consider that `StreamWriter` is used both by the http client and server. If I pass the `traces` in `__init__()` I will have to make something like:
```
def write( ):
    for trace in self.traces:
        await trace.sent_request_chunk_sent(...)
    ...
```
Now if you consider that code in the context of usage by http server, it clearly stops making sense....
This is why I introduced a signal and names it `on_chunk_sent`. Without specifying if it's http request it's sending or http response (as a server).

Reply:
Let's avoid signals for internal-only usage.
Here we have two options: pass a callback for sending signals or derive a `ClientStreamWriter` class. Personally I prefer the second approach.

Reply:
I will remove the `Signal` in favor of accepting an optional callable to be used as a callback. 
I prefer not to create a child class for this - too many layers of inheritance hurts code readability.

Reply:
Ok